sdk/search/search/src/searchServiceClient.ts

```diff
@@ -129,10 +148,57 @@ export class SearchServiceClient {
   public async listIndexes(options: ListIndexesOptions = {}): Promise<Index[]> {
     const { span, updatedOptions } = createSpan("SearchServiceClient-listIndexes", options);
     try {
-      const result = await this.client.indexes.list(
-        operationOptionsToRequestOptionsBase(updatedOptions)
-      );
-      return result.indexes.map(this.generatedIndexToPublicIndex);
+      const result = await this.client.indexes.list({
+        ...operationOptionsToRequestOptionsBase(updatedOptions),
+        select: updatedOptions.select?.join(",")
+      });
+      return result.indexes.map(this.generatedIndexToPublicIndex.bind(this));
```

Comment:
This line needs a bind. So, modified this line and the corresponding lines in listSynonymMaps & listSkillsets method too.

Reply:
Hmm I think I'd prefer to move all the converter methods out of the class and into the module scope so we can avoid needing to bind `this`. That is to say, make all of these helpers standalone non-exported functions below the class body. I'd also be fine moving them into their own separate module.

Reply:
I have moved all of them to separate module. But, out of curiosity, why are we doing this? Is binding this in 2 statements that bad? I am just asking for the purpose of learning. I thought of them as a valid feature. Are they really that bad?

Reply:
The bind isn't bad, it's more that the helpers didn't need to be on the class and extracting them makes it easier to unit test them later. It also cleans up the source for the class significantly. I had just been a little lazy and put them as private methods initially for convenience, but as they grew and multiplied, it seemed like a good time to refactor.

There is a *small* perf hit with the bind, since it makes a new bound copy each time the function is executed, but I don't think that would be enough to really matter in normal usage.